src/main/scala/com/microsoft/hyperspace/actions/RefreshDeleteAction.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.actions
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types.{DataType, StructType}
+
+import com.microsoft.hyperspace.HyperspaceException
+import com.microsoft.hyperspace.actions.Constants.States.{ACTIVE, REFRESHING}
+import com.microsoft.hyperspace.index._
+import com.microsoft.hyperspace.index.DataFrameWriterExtensions.Bucketizer
+import com.microsoft.hyperspace.telemetry.{AppInfo, HyperspaceEvent, RefreshActionEvent}
+
+class RefreshDeleteAction(
```

Comment:
Half of the code is duplicated from `RefreshAction`? Should we refactor? How do you see this design evolving when we introduce different modes? cc: @apoorvedave1

Reply:
It is WIP according to our older discussion to make progress for both Append and Delete. Below are specific decisions I am referring to:

1. (for now) add a new Action to handle delete, later it will be merged to refresh,
2. (for now) protect the feature under a Spark config flag (rather than changing the refresh API).

The main purpose is to make sure the overall approach is correct (i.e. using lineage to detect deleted records and refresh index without scanning original data).

@imback82 @rapoth I can move on to making changes to refresh API and adding mode etc if you are fine with the delete approach here.

Reply:
Yea, let's move forward and iterate on the non-WIP PR.

Reply:
@imback82 I added `refreshActionBase` to capture the common code. The PR is no loner WIP.